wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/Utils.java

```diff
@@ -37,12 +38,8 @@ public static String getDevMachineName(Environment envConfig) {
         for (Map.Entry<String, ? extends MachineConfig> entry : envConfig.getMachines().entrySet()) {
             // TODO should we use server ref instead of installers?
             List<String> installers = entry.getValue().getInstallers();
-            if (installers != null) {
-                for (String installerKey : installers) {
-                    if (installerKey.equals(WSAGENT_INSTALLER) || installerKey.startsWith(WSAGENT_INSTALLER + ":")) {
```

Comment:
I see that it is implemented per-installer, that causes my comment in fact.
The question is why ("you see no reason to change that")?
@tolusha @garagatyi